File: ui/src/app/shared/models/use-case-response.dto.ts
export interface UseCaseNotificationDTO {
    type: string;
    message: string;
    code?: number;
    action?: string;
    reason?: string;
    details?: string[];
    name?: string;
}

export interface UseCaseResponseDTO<T> {
    notifications?: UseCaseNotificationDTO[];
    data?: T;
    _links?: Record<string, string>;
}

File: backend/receiving/src/main/java/com/arcone/biopro/distribution/receiving/adapter/in/web/dto/UseCaseResponseDTO.java
package com.arcone.biopro.distribution.receiving.adapter.in.web.dto;

import java.io.Serializable;
import java.util.List;
import java.util.Map;

public record UseCaseResponseDTO<T>(
    List<UseCaseNotificationDTO> notifications,
    T data,
    Map<String, String> _links
) implements Serializable {
}

